template/src/components/Input/Input.tsx

```diff
@@ -82,7 +81,11 @@ const InputComponent: ForwardRefRenderFunction<InputRef, InputComponentProps> =
         />
 
         {callToAction && (
-          <TouchableOpacity style={{ marginRight: 8 }} onPress={() => callToAction()}>
+          <TouchableOpacity
+            testID='touchable-call-to-action'
```

Comment:
Não faria mais sentido o testID vir através de **props**? Caso a prop testID do próprio Input já esteja sendo passada para o próprio **Input**, uma prop a mais chamada `callToActionTestID` pode ser criada, no caso.

Reply:
Ou podemos reutilizar a testID do próprio input, fazendo o seguinte: <TouchableOpacity testID={`${testID}-call-to-action`} />. Pelo menos é um pattern que mais vejo sendo utilizado e evita criar props para testID, coisa que ficaria inviável dependendo da complexidade do componente. O que achas?

Reply:
Esta de complexiddade não faz sentido. Já que, a prop de teste do callToAction seria pura e simplesmente para o teste do callToAction, e não seria necessário o código estático da forma como você colocou de ter uma string com um texto "chapado", mas esta abordagem aí já é menos pior do que está

Reply:
Em complexidade eu digo, por exemplo, tu faz a componentização de uma seção de uma tela, que será utilizada repetidas vezes em uma mesma tela. Assim, sempre vai ter que repassar todos os testID por props, para cada seção componentizada, múltiplas vezes, caso contrário no teste de integração, o `getByTestId` somente irá retornar o primeiro encontrado. Enquanto com esse pattern de reutilizar, você simplesmente passaria uma única vez e não prejudicaria a legibilidade daquela tela

Reply:
Não estou tratando complexididade, mas sim, objetividade.
Não faz sentido ter código chapado misturado com props pra comportamentos de testID, justamente por serem comportamentos únicos e de certa formas "diferentes" mesmo fazendo parte do mesmo componente.
Então recomendo que haja duas props para que isso sera diferentes